app/services/trade_execution.py

```
@@ -721,76 +722,487 @@ async def _execute_real_order(
+        try:
+            async for db in get_database():
+                from sqlalchemy import select, and_
+                
+                # Get user's active exchange accounts
+                stmt = select(ExchangeAccount, ExchangeApiKey).join(
+                    ExchangeApiKey, ExchangeAccount.id == ExchangeApiKey.account_id
+                ).where(
+                    and_(
+                        ExchangeAccount.user_id == user_id,
+                        ExchangeAccount.status == "active",
+                        ExchangeAccount.trading_enabled == True,
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `ExchangeAccount.trading_enabled:` for truth checks

Replace with `ExchangeAccount.trading_enabled`

(E712)